# Call pm-powersave, so that packages which ship /etc/pm/power.d/ scripts continue to work.
# Taken from upstream git head:
# http://cgit.freedesktop.org/DeviceKit/DeviceKit-power/commit/?id=cad70317771fa6ce1f52af9d357165d47e971d7d
# http://cgit.freedesktop.org/DeviceKit/DeviceKit-power/commit/?id=e808a5063bd4819ca499dcf114371299da88c665
# http://cgit.freedesktop.org/DeviceKit/DeviceKit-power/commit/?id=269fc33a3bb1f8144439cc4c98543510708ab251

Index: devicekit-power/src/dkp-daemon.c
===================================================================
--- devicekit-power.orig/src/dkp-daemon.c	2009-07-01 11:02:09.000000000 +0200
+++ devicekit-power/src/dkp-daemon.c	2009-07-01 11:02:22.000000000 +0200
@@ -448,6 +448,32 @@
 }
 
 /**
+ * dkp_daemon_set_pmutils_powersave:
+ *
+ * Uses pm-utils to run scripts in power.d
+ **/
+static gboolean
+dkp_daemon_set_pmutils_powersave (DkpDaemon *daemon, gboolean powersave)
+{
+	gboolean ret;
+	gchar *command;
+	GError *error = NULL;
+
+	/* run script from pm-utils */
+	command = g_strdup_printf ("/usr/sbin/pm-powersave %s", powersave ? "true" : "false");
+	egg_debug ("excuting command: %s", command);
+	ret = g_spawn_command_line_async (command, &error);
+	if (!ret) {
+		egg_warning ("failed to run script: %s", error->message);
+		g_error_free (error);
+		goto out;
+	}
+out:
+	g_free (command);
+	return ret;
+}
+
+/**
  * gpk_daemon_device_changed:
  **/
 static void
@@ -472,6 +498,9 @@
 		daemon->priv->on_battery = ret;
 		egg_debug ("now on_battery = %s", ret ? "yes" : "no");
 		g_signal_emit (daemon, signals[CHANGED_SIGNAL], 0);
+
+		/* set pm-utils power policy */
+		dkp_daemon_set_pmutils_powersave (daemon, daemon->priv->on_battery);
 	}
 	ret = dkp_daemon_get_low_battery_local (daemon);
 	if (ret != daemon->priv->low_battery) {
@@ -875,6 +904,9 @@
 				    !dkp_daemon_get_on_ac_local (daemon));
 	daemon->priv->low_battery = dkp_daemon_get_low_battery_local (daemon);
 
+	/* set pm-utils power policy */
+	dkp_daemon_set_pmutils_powersave (daemon, daemon->priv->on_battery);
+
 	return daemon;
 }
 
